lib/progress_bar.rb

```diff
@@ -78,6 +83,11 @@ def to_s
     end.strip
   end
 
+  def log_s(text)
```

Comment:
Seems like this might have problems if the text contains newlines.

```ruby
puts "hello\rhi\nfoo"
hillo
foo
```

What about:

```ruby
  def log(text)
    print "\r" + " " * terminal_width + "\r"
    puts text
  end
```

Then you also don't need those other helpers.

Reply:
Im testing `log_s` because I didnt want to hook into `STDOUT` in a spec